packages/react-components/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import cx from 'clsx';
+import {
+  Search as SearchIcon,
+  Close,
+} from '@livechat/design-system-icons/react/material';
+import { Icon, IconTypeName } from '../Icon';
+import { Loader } from '../Loader';
+
+import styles from './Search.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+
+const baseClass = 'search-input';
+const inputBaseClass = `${baseClass}__input`;
+
+export const enum SearchSize {
+  Compact = 'compact',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export interface ISearchProps {
+  isControlledSubmit?: boolean;
+  isCollapsable?: boolean;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  placeholder?: string;
+  size?: SearchSize;
+  value?: string;
+  onChange: (value: string) => void;
+}
+
+export const Search: React.FC<ISearchProps> = ({
+  isControlledSubmit,
+  isCollapsable,
+  isDisabled,
+  isLoading,
+  placeholder = 'Search ...',
+  size = SearchSize.Medium,
+  value,
+  onChange,
+}) => {
+  const [searchValue, setSearchValue] = React.useState<string>(value || '');
+  const [isCollapsed, setIsCollapsed] = React.useState<boolean>(true);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const mergedClassNames = cx(
+    styles[`${inputBaseClass}`],
+    styles[`${inputBaseClass}--${size}`],
+    (isDisabled || isLoading) && styles[`${inputBaseClass}--disabled`],
+    isCollapsable && styles[`${inputBaseClass}--collapsable`],
+    !isCollapsed && styles[`${inputBaseClass}--collapsable--open`]
+  );
+
+  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
+
+    setSearchValue(value);
+    return !isControlledSubmit && onChange(value);
+  };
+
+  const handleOnClear = () => {
+    setSearchValue('');
+    return onChange('');
+  };
+
+  const handleOnClick = () => {
+    if (!isCollapsable) {
+      return;
+    }
+
+    inputRef.current?.focus();
+    return setIsCollapsed(false);
+  };
```

Comment:
The collapsible search can only be expanded once, and then it never collapses again. IMO if we give an option to use collapsible input, it should collapse again when it is not focused and has no input value.

Additionally, the current implementation expands the search input on click, so it is not supporting keyboard navigation. Relying on focus/blur events, you can handle both: pointer and keyboard navigation 🙂 


https://user-images.githubusercontent.com/9700566/169520291-c422f395-9459-4b9e-8dbe-74142c915685.mov


```suggestion
  const handleOnClick = () => {
    inputRef.current?.focus();
  };

  const handleOnFocus = () => {
    if (isCollapsable) {
      setIsCollapsed(false);
    }
  };

  const handleOnBlur = () => {
    if (isCollapsable && !searchValue) {
      setIsCollapsed(true);
    }
  };
```